rfcs/2021-06-15-shipping-blade.md

```diff
@@ -460,6 +466,177 @@ One common question you might have: If exports is not supported by all the tools
 
 The answer to that is because in new versions of webpack(5.x.x) it’s supported. Plus all the new tools are supporting it. Node already supports it. So we want to support modern tools but also have escape hatch/fallback for legacy tools until they upgrade(because we can’t guarantee what versions consumer apps are using)
 
+## Generating TS types for consumers
+I wish generating types was as simple as bundling our source code with rollup. But unfortunately it isn't. There are multiple steps involved for generating types which is of same structure as our output bundle.
+
+Here's the visual representation of the entire process
+![Generating Types](./images/generating-types.png)
+
+Let's see the step by step process of generating types. 
+
+### Phase 1
+First let's begin by looking at our source code. Now if you look at it we have custom declarations files which we also want to be bundled in the output.
+```
+blade/
+└── src/
+    ├── components/
+    │   ├── ThemeProvider/
+    │   │   └── index.ts
+    │   └── index.ts
+    ├── tokens/
+    │   ├── theme/
+    │   │   ├── paymentTheme.ts
+    │   │   └── theme.d.ts
```

Comment:
Are we thinking of these `d.ts` files incorrectly or something different from a TS design principle? Can we think of alternate ways to achieve this?

Let's discuss on Slack.